### PR TITLE
Add e2e test for service annotation 'ServiceAnnotationLoadBalancerResourceGroup'

### LINKED ISSUE #184: Add e2e tests for service annotation "service.beta.kubernetes.io/azure-load-balancer-resource-group"
Add e2e tests for service annotation "service.beta.kubernetes.io/azure-load-balancer-resource-group".

/kind testing
/milestone v1.16

### PR DESCRIPTION
**Pull request by nilo19**
Add e2e test for service annotation `ServiceAnnotationLoadBalancerResourceGroup`

Fixes #184 

/sig testing

**Release note**:
```
None
```

### CODE REVIEW

File: tests/e2e/utils/resource_group_utils.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+		_, err := gc.Delete(context.Background(), rgName)
+		Expect(err).NotTo(HaveOccurred())
+		return
+	}
+}
+
+func createTestTemplate(name *string) resources.Group {
+	return resources.Group{
+		Name:     name,
+		Location: to.StringPtr("eastasia"),
```
Comment (feiskyer) [inline tests/e2e/utils/resource_group_utils.go]: We shouldn't use a fixed location here. Seems the location is not part of e2e configure file, so let's get it from node label `failure-domain.beta.kubernetes.io/region`.

If there're something wrong when getting this, then use a default location `eastus2`, which is same as test-infra configs.

Comment (nilo19) [reply tests/e2e/utils/resource_group_utils.go]: Got it.

**Comment by feiskyer** [comment]
/lgtm
/approve

thanks

**Pull request merged**